crates/chain/src/tx_graph.rs

```diff
@@ -1,12 +1,25 @@
 //! Module for structures that store and traverse transactions.
 //!
-//! [`TxGraph`] is a monotone structure that inserts transactions and indexes the spends. The
-//! [`ChangeSet`] structure reports changes of [`TxGraph`] but can also be applied to a
-//! [`TxGraph`] as well. Lastly, [`TxDescendants`] is an [`Iterator`] that traverses descendants of
-//! a given transaction.
+//! [`TxGraph`] is a monotone structure that contains transactions and indexes the spends. It can
```

Comment:
I felt this bit needed a bit of filling out. May need integration into surrounding.

```suggestion
//! [`TxGraph`] contains transactions and indexes them so you can easily traverse the graph of those transactions.
`TxGraph` is *monotone* in that you can always insert a transaction -- it doesn't care whether that transaction is in the current best chain or whether it conflicts with any of the existing transactions or what order you insert the transactions.
Furthermore, there is currently no way to delete a transaction.
```